### PR TITLE
[GNA] Introduce an option to invoke the QoS feature

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Introduce an option to invoke the QoS feature
 - Clean unused `propagateSync` function
### Tickets:
 - 56759

### CODE REVIEW

**Review by sirzabek — APPROVED** [review]

File: inference-engine/samples/speech_sample/speech_sample.hpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ static const char model_message[] = "Required. Path to an .xml file with a train
 
 /// @brief message for assigning cnn calculation to device
 static const char target_device_message[] = "Optional. Specify a target device to infer on. CPU, GPU, MYRIAD, GNA_AUTO, GNA_HW, "
-                                            "GNA_SW_FP32, "
+                                            "GNA_HW_WITH_SW_FBACK, GNA_SW_FP32, "
                                             "GNA_SW_EXACT and HETERO with combination of GNA as the primary device and CPU"
                                             " as a secondary (e.g. HETERO:GNA,CPU) are supported. The list of available devices is shown "
                                             "below. "
```
[the remark this answers is not in the document]
Comment (kbruniec) [reply inference-engine/samples/speech_sample/speech_sample.hpp]: Agree. Although this is outside of this PR changes, I will remove this sentence.

**Comment by nkogteva** [comment]
Basically, I'm against having this in performance counters in such form. 

Performance counters are general functionality. It is GNA plugin's problem that the current implementation is not aligned with other plugins. But anyway I would like to avoid more misalignments. 

First of all, performance counters are contain 2 numbers for each position: realTime and cpuTime. If realTime is not empty, that means that a primitive was executed on the device. For CPU plugin these numbers are equal, but for other plugins, it is clear whether hw was used or not. 

Secondly, the performance counter is reported per request. So it is supposed that numbers are averaged per several infers whether the same request is used. So we can't report 1 or 0 in counters if behavior can be different for different infers on the same request.

Could you please explain the goal of this new feature in order to decide how better to implement it?

**Comment by kbruniec** [comment]
@nkogteva Thanks for your valuable comment (especially explanation for 1realTime1). I got your point and agree.
I've pushed a commit to address your concerns. Please take a look if it's OK now.

File: inference-engine/samples/speech_sample/main.cpp [hunk 2]
```diff
@@ -346,16 +349,21 @@ void getPerformanceCounters(InferenceEngine::InferRequest& request, std::map<std
...
  * @return none.
  */
 void sumPerformanceCounters(std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> const& perfCounters,
-                            std::map<std::string, InferenceEngine::InferenceEngineProfileInfo>& totalPerfCounters) {
+                            std::map<std::string, InferenceEngine::InferenceEngineProfileInfo>& totalPerfCounters,
+                            uint64_t& totalRunsOnHw) {
+    auto runOnHw = false;
     for (const auto& pair : perfCounters) {
         totalPerfCounters[pair.first].realTime_uSec += pair.second.realTime_uSec;
+        runOnHw |= pair.second.realTime_uSec > 0; // if realTime is above zero, that means that a primitive was executed on the device
```
Comment (mryzhov) [inline inference-engine/samples/speech_sample/main.cpp:364]:
```suggestion
        runOnHw |= pair.second.realTime_uSec > 0;  // if realTime is above zero, that means that a primitive was executed on the device
```

**Review by nkogteva — APPROVED** [review]

**Pull request merged**